### ISSUE TITLE
Country Limitations: allow separate rules for billing vs shipping

### ISSUE THREAD

**Issue by drbyte** [zencart/zencart#3649]
Currently if a Country's `status` is disabled, then any customer who has an address-book entry with that country in it will be stopped in their tracks until they delete that address or edit it to another country.

In the case of stores who will ship to an address other than the billing address, (eg: US merchant ships to a US address near the border of Canada or Mexico and the customer makes their own arrangements for pickup/forwarding, at their own cost and their own duties arrangements), then it would be beneficial for the storeowner to enable such capability.

**Comment by lat9** [comment]
How about we start by adding another field to the `countries` table that works in conjunction with the currently-defined `status`?  Let's name it `status_limitation` (until a better name is identified).

`countries::status_limitation` will be a single-character field, containing one of:
- **B** ... the default, indicating that the country is valid for **B**oth shipping and billing addresses.
- **S** ... indicates that the country is valid for **S**hipping only.
- **B** ... indicates that the country is valid for **B**illing only.

Changes will be required for (at a minimum);
1. The admin's **_Locations/Taxes::Countries_** tool
2. The storefront checkout-processing
    a. `checkout_shipping`.  Verify that the current ship-to address is valid for shipping.
    b. `checkout_payment`. Verify that the current bill-to address is valid for billing.

Should the update include some type of messaging to the customer, perhaps noting (via database lookup) that there are Billing and/or Shipping limitations currently defined by the site?

**Comment by drbyte** [comment]
Choices `B / S / B` ... probably can't use `B` twice. 😄 
Maybe `A` for "All" instead of "B" for "Both" if "B" is also for "Billing"?

I feel like a new field makes the existing `status` field redundant. 

In reality I expect most will be "N: Not Supported".
Next most will be "All"/Both
Next most will be "Billing only"
Rarely will "Shipping only" be used IMO.

Aside: `Naming is hard!`: Why do "Ship/Send" and "Sell" have to all start with S?!! And "Bill" and "Both" with B?!!!  And `I` for Invoice looks like `1` or `l`. 🙄

**Comment by lat9** [comment]
Duh (on the B and B).  On the `status` redundancy, I agree ... but keeping downward compatibility says that it stays, IMO.  I also agree that the Shipping-only will be rarely used, I included that for completeness.

**Comment by scottcwilson** [comment]
You could spring for an additional 3 characters and use "Bill", "Ship" and "Both" - more self documenting.

**Comment by drbyte** [comment]
> Should the update include some type of messaging to the customer, perhaps noting (via database lookup) that there are Billing and/or Shipping limitations currently defined by the site?

Yes, the existing messaging in `init_includes/init_header.php` (which shows up on the address-book-edit screen that it redirects to) would need to be made more friendly.

I've added this filename to your prior post above.

**Comment by lat9** [comment]
Another 'complication': `paypalwpp`.  Need to ensure that any address-override coming from PayPal is, in fact, supported by the store's limitations.
